pkg/kudoctl/packages/resolver/resolver.go

```diff
@@ -49,32 +49,62 @@ func NewInClusterResolver(c *kudo.Client, ns string) Resolver {
 // For local access there is a need to provide absolute or relative path as part of the name argument. `cassandra` without a path
 // component will resolve to the remote repo.  `./cassandra` will resolve to a folder which is expected to have the operator structure on the filesystem.
 // `../folder/cassandra.tgz` will resolve to the cassandra package tarball on the filesystem.
-func (m *PackageResolver) Resolve(name string, appVersion string, operatorVersion string) (p *packages.Resources, err error) {
+func (m *PackageResolver) Resolve(name string, appVersion string, operatorVersion string) (*packages.FancyResources, error) {
 
-	// Local files/folder have priority
-	_, err = m.local.fs.Stat(name)
-	// force local operators usage to be either absolute or express a relative path
-	// or put another way, a name can NOT be mistaken to be the name of a local folder
-	if filepath.Base(name) != name && err == nil {
-		var abs string
-		abs, err = filepath.Abs(name)
-		if err != nil {
-			return nil, err
-		}
+	clog.V(1).Printf("determining package type of %v", name)
+
+	// 1. local files/folder have priority
+	abs, err := m.local.LocalPackagePath(name)
+	if err == nil {
 		clog.V(2).Printf("local operator discovered: %v", abs)
-		p, err = m.local.Resolve(name, appVersion, operatorVersion)
-		return
+
+		var res *packages.Resources
+		if strings.HasSuffix(abs, ".tgz") {
```

Comment:
Nit: Should we support ".tar.gz" suffix as well? Though the existing code already required ".tgz" in the local resolver.

Reply:
Hmmm... We could, I guess... Although the packages should generally be created by KUDO, right? I would say let's keep tgz only for now, maybe we can add it later. It might require changes at different places.